kafka-rest/src/main/java/io/confluent/kafkarest/entities/GetTopicPartitionOffsetResponse.java

```diff
@@ -0,0 +1,143 @@
+import java.util.Optional;
+import javax.annotation.Nullable;
+
+/**
+ * Response for GET /topics/(topic)/partitions/(partition)/offsets requests.
+ *
+ * <p>It will either be:</p>
+ *
+ * <ul>
+ *   <li>a summary containing the beginning and end offsets</li>
+ *   <li>an offset and its associated timestamp</li>
+ *   <li>empty</li>
+ * </ul>
+ *
+ * @see io.confluent.kafkarest.resources.v2.PartitionsResource#getOffsets(String, int, Instant)
+ */
+public final class GetTopicPartitionOffsetResponse {
+
+  private static final GetTopicPartitionOffsetResponse EMPTY =
+      new GetTopicPartitionOffsetResponse(
+          /* beginning= */ null, /* end= */ null, /* offset= */ null, /* timestamp= */ null);
+
+  private final @Nullable Long beginning;
+  private final @Nullable Long end;
+  private final @Nullable Long offset;
+  private final @Nullable Instant timestamp;
+
+  private GetTopicPartitionOffsetResponse(
+      @Nullable Long beginning,
+      @Nullable Long end,
+      @Nullable Long offset,
+      @Nullable Instant timestamp
+  ) {
+    this.beginning = beginning;
+    this.end = end;
+    this.offset = offset;
+    this.timestamp = timestamp;
+  }
+
+  /**
+   * Returns a response with a summary containing the {@code beginning} and {@code end} offsets.
+   */
+  public static GetTopicPartitionOffsetResponse createSummary(long beginning, long end) {
+    return new GetTopicPartitionOffsetResponse(
+        beginning, end, /* offset= */ null, /* timestamp= */ null);
+  }
+
+  /**
+   * Returns a response with the given {@code offset} and its associated {@code timestamp}.
+   */
+  public static GetTopicPartitionOffsetResponse createForTime(long offset, Instant timestamp) {
+    return new GetTopicPartitionOffsetResponse(
+        /* beginning= */ null, /* end= */ null, offset, requireNonNull(timestamp, "timestamp"));
+  }
+
+  /**
+   * Returns an empty response.
+   */
+  public static GetTopicPartitionOffsetResponse empty() {
+    return EMPTY;
+  }
+
+  /**
+   * The earliest offset in the topic partition.
+   */
+  @JsonProperty(value = "beginning", access = Access.READ_ONLY)
+  @JsonInclude(Include.NON_ABSENT)
+  public Optional<Long> getBeginning() {
+    return Optional.ofNullable(beginning);
+  }
+
+  /**
+   * The latest offset in the topic partition.
+   */
+  @JsonProperty(value = "end", access = Access.READ_ONLY)
+  @JsonInclude(Include.NON_ABSENT)
+  public Optional<Long> getEnd() {
+    return Optional.ofNullable(end);
+  }
+
+  /**
+   * A specific offset in the topic partition.
+   */
+  @JsonProperty(value = "offset", access = Access.READ_ONLY)
+  @JsonInclude(Include.NON_ABSENT)
+  public Optional<Long> getOffset() {
+    return Optional.ofNullable(offset);
+  }
+
+  /**
+   * The timestamp associated with {@link #getOffset()} in the topic partition.
+   */
+  @JsonProperty(value = "timestamp", access = Access.READ_ONLY)
+  @JsonInclude(Include.NON_ABSENT)
+  public Optional<Instant> getTimestamp() {
```

Comment:
I'm personally inclined towards using `long` for timestamp because: 

>  1. consistency with the java API. 2. the meaning of timestamp in kafka is almost arbitrary, so it's conceivable people will not want a standard interpretation of it. 3. more definitely easy to parse by all client languages (also, it's well under the javascript max int so no problems there). 4. unix time is not obscure, so not really confusing or hard to deal with in any language.

Anyone else feel free to give an opinion / reasons here. I'm happy to accept either way, just stating a preference (which is not strongly held, i'd be easily convinced the other way).